.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,64 @@
+name: Deploy
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    name: Build with Vite
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Setup Node.js environment
+        uses: actions/setup-node@v2
+        with:
+          node-version: '18'
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Build
+        run: npm run build
+
+      - name: Archive production artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: app-build
+          path: dist/*
+
+  deploy:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Download artifacts
+        uses: actions/download-artifact@v2
+        with:
+          name: app-build
+          path: dist/
+
+      - name: Copy files
+        uses: appleboy/scp-action@master
+        with:
+          host: ${{ secrets.EC2HOSTNAME }}
```

Comment:
I think this is an AWS side thing but wanted to verify: we'll need to add an environment variable API_BASE_URL that stores the base url for the live backend.

Reply:
I think we could set it as an environment variable on the server, but I'd honestly prefer to just add it to the front end src code, especially at least for right now. The API is secured with an API key, doing it this way either makes the deployment more complex or relies on me to manually go into server and update env variables, this app is not critical infrastructure, and if we were going to be serious about securing the private api route, we'd need to go through and update things so that it's not exposed on swagger-ui.